compiler/p/codegen/FPTreeEvaluator.cpp

```diff
@@ -284,24 +292,35 @@ TR::Register *OMR::Power::TreeEvaluator::dbits2lEvaluator(TR::Node *node, TR::Co
       {
       TR::Register *condReg = cg->allocateRegister(TR_CCR);
 
-      generateTrg1Src2Instruction(cg, TR::InstOpCode::fcmpu, node, condReg, doubleReg, doubleReg);
-      TR::PPCControlFlowInstruction *cfop = (TR::PPCControlFlowInstruction *)
-              generateControlFlowInstruction(cg, TR::InstOpCode::cdnan, node);
+      TR::LabelSymbol *nanNormalizeStartLabel = generateLabelSymbol(cg);
+      TR::LabelSymbol *nanNormalizeEndLabel = generateLabelSymbol(cg);
 
+      uint16_t numDeps = TR::Compiler->target.is64Bit() ? 2 : 3;
+      TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(numDeps, numDeps, cg->trMemory());
+      TR::addDependency(deps, condReg, TR::RealRegister::NoReg, TR_CCR, cg);
+      if (TR::Compiler->target.is64Bit())
+         {
+         TR::addDependency(deps, lReg, TR::RealRegister::NoReg, TR_GPR, cg);
+         }
+      else
+         {
+         TR::addDependency(deps, lowReg, TR::RealRegister::NoReg, TR_GPR, cg);
+         TR::addDependency(deps, highReg, TR::RealRegister::NoReg, TR_GPR, cg);
+         }
+
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::fcmpu, node, condReg, doubleReg, doubleReg);
+      generateLabelInstruction(cg, TR::InstOpCode::label, node, nanNormalizeStartLabel);
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::beq, PPCOpProp_BranchLikely, node, nanNormalizeEndLabel, condReg);
       if (TR::Compiler->target.is64Bit())
          {
-         cfop->addTargetRegister(lReg);
-         cfop->addSourceRegister(condReg);
-         cfop->addSourceRegister(lReg);
+         loadConstant(cg, node, (int64_t)CONSTANT64(0x7ff8000000000000), lReg);
```

Comment:
Hope loadConstant will do a good job of this, generating two instructions:   lis lReg, 0x7ff8;  rldicr lReg,lReg,32,32
Otherwise, you can improve this.

Reply:
It looks like our current implementation of `loadConstant` doesn't perform this optimization for whatever reason. Currently, it seems to always try to use the pTOC for any values that are outside the range of a 32-bit signed integer.

If that's an optimization we want, then in my opinion the proper course of action would be to open an issue to have the `loadConstant` function modified to support doing that.